modules/module/LoRAModule.py

```diff
@@ -1,66 +1,76 @@
+import abc
+import enum
+import itertools
 import math
-from abc import ABCMeta
+import typing
 
 import torch
 from torch import nn, Tensor
 from torch.nn import Linear, Conv2d, Parameter
 
+ModuleDict = dict[str, nn.Module]
+StateDict = dict[str, torch.Tensor]
 
-class LoRAModule(metaclass=ABCMeta):
-    prefix: str
-    orig_module: nn.Module
-    lora_down: nn.Module
-    lora_up: nn.Module
-    alpha: torch.Tensor
+class LoRAType(enum.Enum):
+    LierLa = enum.auto()
```

Comment:
the LoRA Type needs to be stored in the TrainConfig, which is serialized as a json file. To enable this, I always defined fixed strings for enums. Please do the same here. Also, All the other enums are saved in modules/util/enum. This one is very specific to the LoRAModuleWrapper, but I still think it should be moved to stay consistent.